process/releases.md

```diff
@@ -0,0 +1,73 @@
+# Releases
+
+By releasing we mean deploying something to production. This won't go into the external governance around doing this, for example raising a *Request for Change*. It will provide a guide on how to prepare git based projects for release.
+
+It also assumes you are following the guidance in [workflows](workflows.md) and [pull requests](pull_request.md) when working on a repo.
+
+## Applications & libraries
+
+By *application* we mean a web application or service that is executed and runs in an environment.
+
+The general principles for releasing these are
+
+- you [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) the application's repo when you have changes you are ready to release
+- you deploy to production using *tags* instead of *branches*
+
+If a library we have created is also published to a package manager such as [npm](https://www.npmjs.com/) or [rubygems](https://rubygems.org/), it should follow these same principles. If its not published but the repo is instead directly linked to, then you can ignore this release process.
+
+### What is a tag
+
+In short a tag is an alias for a commit hash, and is used to mark specific points in a repo's git history. Typically its used to mark release points hence its use here.
+
+### Creating a tag
+
+There are 2 options for creating a tag
+
+- manually using git
+- via GitHub when creating a release
+
+#### Manually Tagging the repo
+
+To tag a repo you first need to create the tag locally and then push it to GitHub, in much the same way as creating a new branch
+
+```bash
+git tag -a v1.4 -m "Version 1.4"
+git push origin v1.4
+```
+
+The argument `v1.4` can actually be whatever you like, but you should follow [semantic versioning](https://semver.org/) when deciding what tag to apply. The `-m "Version 1.4"` is simply a message or description you can store with the tag. If you omit this argument git will display your default editor instead, just as if you were adding a commit message.
+
+With the repo tagged and the new tag pushed to GitHub, its now ready to be deployed.
+
+It's also worth noting you can still create a [GitHub release](#github-releases) based on a tag you have created manually.
+
+#### GitHub releases
```

Comment:
Are we actually doing this?  If so, who are they for?

It would be good to put something into this guidance explaining why we do or don't use them.

I'm just a little bit wary of clogging up the source code repository with binary release files.  It's fine if there's a good reason for it but it would be good to understand that.

Reply:
🤔 I hadn't considered that. Though my quick googling suggests these (i.e. releases) are a GitHub only thing and don't live in git. So when you clone a repo you are not downloading any binaries.

I create releases on my projects for the specific reason of capturing when a version was pushed to production and under what RfC. This is because I create the tag first, then we deploy to pre-prod using that tag as a double-check, before then raising the RfC and waiting to ship.

If I didn't have the RfC bit in the middle, I could capture all that info in the tag itself. But for now, releases give me a nice record of when projects were actually shipped to production, and also can tell you what is currently deployed.

<img width="755" alt="screen shot 2018-09-28 at 09 39 01" src="https://user-images.githubusercontent.com/1789650/46197718-7c720c00-c302-11e8-99eb-f284d3d9200c.png">

So this screenshot hopefully conveys the version we have in production and when we deployed it, along with the fact we have a new one ready to go.

I could reword this into a reason for using releases. However wanting to KISS, perhaps I should drop the section on GitHub releases and just spread the word for now???